components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -332,11 +340,31 @@ private void triggerNotification(User user, String type, String code, Property[]
         }
         properties.put(IdentityRecoveryConstants.TEMPLATE_TYPE, type);
         Event identityMgtEvent = new Event(eventName, properties);
+
+        String action;
+        if (recoveryDataDO !=null && recoveryDataDO.getRecoveryScenario()
```

Comment:
```suggestion
        if (recoveryDataDO != null && recoveryDataDO.getRecoveryScenario()
```

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/452/commits/db3db432beb65c5e6174862949a01252d2f26d94